scraping/script.ipynb

```diff
@@ -19,25 +19,103 @@
+    "\n",
+    "# Dictionary mapping Unicode fraction symbols to their decimal equivalents\n",
+    "fraction_mapping = {\n",
+    "    '\\u00BD': '1/5',  # 1/2\n",
+    "    '\\u00BC': '1/4', # 1/4\n",
+    "    '\\u00BE': '3/4', # 3/4\n",
+    "    '\\u2153': '1/3', # 1/3\n",
+    "    '\\u2154': '2/3', # 2/3\n",
+    "    '\\u215b': '1/8', # 1/8\n",
+    "    '\\u00b0': 'degrees ', #°\n",
+    "    '\\u2013': '-', # -\n",
+    "    '\\u00a0': ' ' # space\n",
+    "    # Add more mappings as needed\n",
+    "}\n",
+    "\n",
+    "def replace_fraction_symbols(text):\n",
+    "    # Define a regex pattern that captures numbers possibly adjacent to fraction symbols\n",
+    "    pattern = r'(\\d*)(%s)(\\d*)' % '|'.join(re.escape(key) for key in fraction_mapping.keys())\n",
+    "\n",
+    "    # Function to replace each match with appropriate spacing\n",
+    "    def replace(match):\n",
+    "        # Pre-number, fraction, and post-number\n",
+    "        pre, frac, post = match.groups()\n",
+    "        # Replace the fraction with its decimal equivalent from the dictionary\n",
+    "        frac_decimal = fraction_mapping[frac]\n",
+    "        # Add space if there is a preceding or succeeding number\n",
+    "        if frac_decimal == 'degrees ' or frac_decimal == '-' or frac_decimal == ' ':\n",
+    "            if pre and post:\n",
+    "                return f'{pre} {frac_decimal} {post}'\n",
+    "            elif pre:\n",
+    "                return f'{pre} {frac_decimal}'\n",
+    "            elif post:\n",
+    "                return f'{frac_decimal} {post}' \n",
+    "        elif pre and post:\n",
+    "            return f'{pre} and {frac_decimal} {post}'\n",
+    "        elif pre:\n",
+    "            return f'{pre} and {frac_decimal}'\n",
+    "        elif post:\n",
+    "            return f'{frac_decimal} {post}'\n",
+    "        return frac_decimal\n",
+    "\n",
+    "    return re.sub(pattern, replace, text)\n",
+    "\n",
+    "\n",
     "## Extracts all recipe URLs from a given page. ##\n",
     "def get_recipes_from_page(url):\n",
     "    response = requests.get(url)\n",
     "    soup = BeautifulSoup(response.text, 'html.parser')\n",
-    "    recipe_links = [a['href'] for a in soup.find_all('a', href=True) if \"/recipes/\" in a['href']]\n",
-    "    return recipe_links\n",
+    "\n",
+    "    # Base URL for relative links\n",
+    "    base_url = \"https://en.wikibooks.org\"\n",
+    "\n",
+    "    # Extract the href attributes, focusing on individual recipe pages\n",
+    "    recipe_links = []\n",
+    "    for link in soup.find_all('a', href=True):\n",
+    "        href = link['href']\n",
+    "        # Look for links that are likely to be individual recipes\n",
+    "        if href.startswith('/wiki/Cookbook:') and not \"Category:\" in href:\n",
+    "            full_url = base_url + href\n",
+    "            recipe_links.append(full_url)\n",
+    "\n",
+    "    return recipe_links[2:-1]\n",
```

Comment:
Maybe just add a little comment specifying why first 2 and last elements of the return list aren't considered.

Reply:
Good point I will :)